plugins/woocommerce/src/Internal/Traits/OrderAttributionMeta.php

```diff
@@ -381,48 +381,29 @@ private function get_field_description( string $field ): string {
 	/**
 	 * Get the order history for the customer (data matches Customers report).
 	 *
-	 * @param mixed $customer_identifier The customer ID or billing email.
+	 * @param int $customer_report_id The customer ID (not necessarily User ID).
 	 *
-	 * @return array Order count, total spend, and average spend per order.
+	 * @return array|null Order count, total spend, and average spend per order.
 	 */
-	private function get_customer_history( $customer_identifier ): array {
-		/*
-		 * Exclude the statuses that aren't valid for the Customers report.
-		 * 'checkout-draft' is the checkout block's draft order status. `any` is added by V2 Orders REST.
-		 * @see /Automattic/WooCommerce/Admin/API/Report/DataStore::get_excluded_report_order_statuses()
-		 */
-		$all_order_statuses = ReportsController::get_order_statuses();
-		$excluded_statuses  = array( 'pending', 'failed', 'cancelled', 'auto-draft', 'trash', 'checkout-draft', 'any' );
-
-		// Get the valid customer orders.
-		$args = array(
-			'limit'  => - 1,
-			'return' => 'objects',
-			'status' => array_diff( $all_order_statuses, $excluded_statuses ),
-			'type'   => 'shop_order',
-		);
-
-		// If the customer_identifier is a valid ID, use it. Otherwise, use the billing email.
-		if ( is_numeric( $customer_identifier ) && $customer_identifier > 0 ) {
-			$args['customer_id'] = $customer_identifier;
-		} else {
-			$args['billing_email'] = $customer_identifier;
-			$args['customer_id']   = 0;
-		}
-
-		$orders = wc_get_orders( $args );
-
-		// Populate the order_count and total_spent variables with the valid orders.
-		$order_count = count( $orders );
-		$total_spent = 0;
-		foreach ( $orders as $order ) {
-			$total_spent += $order->get_total() - $order->get_total_refunded();
-		}
+	private function get_customer_history( $customer_report_id ): ?array {
+		$matching_customers = $this->get_customer_reports_rest_response( array( 'customers' => $customer_report_id ) );
```

Comment:
```suggestion
		$matching_customers = wc()->api->get_endpoint_data( '/wc-analytics/reports/customers',  array( 'customers' => $customer_report_id );
```
I think that this should do the trick and it allows you to get rid of `get_customer_reports_rest_response` function.